File: scripts/parity/scenarios/js/arc.mjs
// Parity scenario: arc using arcXY geometry function
import { Point, arcXY, transform } from '../../../../dist/index.js'

// Create an arc using arcXY
const centre_point = new Point({ x: 50, y: 50, z: 0.2 })
const radius = 10
const start_angle = 0
const arc_angle = 0.75 * Math.PI * 2  // 0.75 * tau
const segments = 64

const steps = arcXY(centre_point, radius, start_angle, arc_angle, segments)

const g = transform(steps, 'gcode', { show_tips: false, show_banner: false }).gcode
process.stdout.write(g.trimEnd() + '\n')

File: scripts/parity/scenarios/py/arc.py
import fullcontrol as fc
from math import tau

# Create an arc using arcXY
centre_point = fc.Point(x=50, y=50, z=0.2)
radius = 10
start_angle = 0
arc_angle = 0.75 * tau
segments = 64

steps = fc.arcXY(centre_point, radius, start_angle, arc_angle, segments)

res = fc.transform(steps, result_type='gcode')
print(res)
